app/src/main/java/tech/ula/utils/AndroidUtility.kt

```diff
@@ -46,12 +47,13 @@ fun arePermissionsGranted(context: Context): Boolean {
                     Manifest.permission.WRITE_EXTERNAL_STORAGE) == PackageManager.PERMISSION_GRANTED)
 }
 
-fun displayGenericErrorDialog(activity: Activity, titleId: Int, messageId: Int) {
+fun displayGenericErrorDialog(activity: Activity, titleId: Int, messageId: Int, callback: (() -> Unit)? = null) {
```

Comment:
I think the `callback` parameter could better follow idiomatic Kotlin by using an empty function default, `{}` and a non-nullable type.